### PR TITLE
Add ability to display a filesystem detail view

### PR DESCRIPTION
**Pull request by mulkieran**
Related https://github.com/stratis-storage/project/issues/510

### CODE REVIEW

**Comment by mulkieran** [comment]
Probably PoolSelector should have an optional field.

**Comment by mulkieran** [comment]
Test in final commit will fail.

**Comment by mulkieran** [comment]
Test should now succeed.

**Comment by mulkieran** [comment]
@bmr-cymru @drckeefe The basic bugs seem to have been worked out. You are welcome to comment on the format of the table and the detail output. Note that snapshot origin information is not available in this PR so do not expect to see it.

**Comment by bgurney-rh** [comment]
Some examples of output, given the following test pools, filesystems, and snapshots:
```
# stratis pool create spool1 /dev/nvme0n1p1
# stratis pool create spool2 /dev/nvme0n1p2
# stratis pool create spool2 /dev/nvme0n1p3
# stratis pool create spool3 /dev/nvme0n1p3
# stratis fs create spool1 sfs1
# stratis fs create spool2 sfs2
# stratis fs create spool3 sfs3
# stratis fs snapshot spool3 sfs3 ssnap3

# stratis fs list spool3 --name sfs3
UUID: 247404b8-7ff4-4769-a106-450e1e6851f7
Name: sfs3
Pool: spool3

Device: /dev/stratis/spool3/sfs3

Created: Jan 18 2024 13:27

Sizes:
  Logical size of devicemapper device: 1 TiB
  Total used (including for XFS metadata): 546 MiB
  Free: 1023.47 GiB
  
  Size Limit: None


# stratis fs list spool3 --name ssnap3
UUID: 7a9683ee-ce12-46f3-a897-83b2d2afadd3
Name: ssnap3
Pool: spool3

Device: /dev/stratis/spool3/ssnap3

Created: Jan 18 2024 15:41

Sizes:
  Logical size of devicemapper device: 1 TiB
  Total used (including for XFS metadata): 546 MiB
  Free: 1023.47 GiB
  
  Size Limit: None

# stratis fs list
Pool     Filesystem   Total / Used / Free / Limit            Device                       UUID                                
spool1   sfs1         1 TiB / 546 MiB / 1023.47 GiB / None   /dev/stratis/spool1/sfs1     498c49b4-1895-4a9b-ac2b-6bfbbb254639
spool2   sfs2         1 TiB / 546 MiB / 1023.47 GiB / None   /dev/stratis/spool2/sfs2     41b56bd6-30f8-46aa-8b93-c247461caf0d
spool3   sfs3         1 TiB / 546 MiB / 1023.47 GiB / None   /dev/stratis/spool3/sfs3     247404b8-7ff4-4769-a106-450e1e6851f7
spool3   ssnap3       1 TiB / 546 MiB / 1023.47 GiB / None   /dev/stratis/spool3/ssnap3   7a9683ee-ce12-46f3-a897-83b2d2afadd3
```

**Comment by mulkieran** [comment]
rebased.

**Comment by mulkieran** [comment]
rebased, conflicts resolved.

**Review by drckeefe — APPROVED** [review]

File: src/stratis_cli/_actions/_list_filesystem.py [hunk 1]
```diff
@@ -69,57 +132,61 @@ def filesystem_size_quartet(dbus_props):
...
+        print(f"UUID: {self.uuid_formatter(fs.Uuid())}")
+        print(f"Name: {fs.Name()}")
+        print(f"Pool: {self.path_to_name[fs.Pool()]}")
+        print()
+        print(f"Device: {fs.Devnode()}")
+        print()
+        print(f"Created: {created}")
+        print()
+        print("Sizes:")
+        print(f"  Logical size of devicemapper device: {total}")
```
[the remark this answers is not in the document]
Comment (mulkieran) [reply src/stratis_cli/_actions/_list_filesystem.py:188]: "thin" is probably better.

File: src/stratis_cli/_actions/_list_filesystem.py [hunk 2]
```diff
@@ -69,57 +132,61 @@ def filesystem_size_quartet(dbus_props):
...
+        print(f"  Total used (including for XFS metadata): {used}")
```
[the remark this answers is not in the document]
Comment (mulkieran) [reply src/stratis_cli/_actions/_list_filesystem.py:189]: It reads worse for me, but I'll do it, since I think it will be understood correctly.

**Review by bgurney-rh — APPROVED** [review]

**Pull request merged**